modules/top-level/output.nix

```diff
@@ -74,15 +74,54 @@ in
 
   config =
     let
+      # Plugin list extended with dependencies
+      allPlugins =
+        let
+          pluginWithItsDeps = p: [ p ] ++ builtins.concatMap pluginWithItsDeps p.dependencies or [ ];
+        in
+        lib.unique (builtins.concatMap pluginWithItsDeps config.extraPlugins);
+
+      # All plugins with doc tags removed
+      allPluginsOverridden = map (
+        plugin:
+        plugin.overrideAttrs (prev: {
+          nativeBuildInputs = lib.remove pkgs.vimUtils.vimGenDocHook prev.nativeBuildInputs or [ ];
+          configurePhase = builtins.concatStringsSep "\n" (
+            builtins.filter (s: s != ":") [
+              prev.configurePhase or ":"
+              "rm -vf doc/tags"
```

Comment:
I think the intention was to avoid the blank line being concatenated.

Using `optional` or filtering null would be better:

```nix
concatStringsSep "\n" (
  optional (prev ? configurePhase) prev.configurePhase
  ++ [ "rm -vf doc/tags" ]
)
```

Reply:
I don't think it's too bad to have empty lines

Reply:
I usually do it this way to avoid empty lines. With your suggestion first line will be empty and it bother me when I look at a derivation with `nix derivation show`. Maybe it can be trimmed with some lib function instead.

Reply:
If you have a simpler way I'd not be against it, but I'd rather go with a bit simpler though annoying

Reply:
I think @MattSturgeon could be good enough

Reply:
> Maybe it can be trimmed with some lib function instead.

https://github.com/NixOS/nixpkgs/pull/315411 doesn't seem to be going anywhere :sob: 

> I usually do it this way to avoid empty lines.

See my above comment if that's important to you, although as I said @traxys' suggestion is simpler.
I don't like the idea of "magic" values like `":"`, though filtering `"" || null` would be ok too.

Reply:
`:` is not a magic string. Unfortunately it comes from the vimPlugins in nixpkgs. From here: https://github.com/NixOS/nixpkgs/blob/ad0b5eed1b6031efaed382844806550c3dcb4206/pkgs/applications/editors/vim/plugins/build-vim-plugin.nix#L18. So I did it long time ago and I didn't like both empty lines and redundant `:` if I just append to whatever is there. vimPlugins from nixpkgs always have configure phase defined with `:` there.

Reply:
Then something like:
```nix
let oldPhase = if prev ? configurePhase && prev.configurePhase != ":" then configurePhase +'\n' else ""; in
   oldPhase + "rf -vf docs/tags";
```
?

It's a bit less confusing I think

Reply:
But I think I'll change it to the

```
configurePhase = ''
    ${prev.configurePhase or  ""}
    rm -vf doc/tags
'';
```

It was from my home config where I often bother myself with such things because of perfectionism. I pretty much distastes many generated in nix configs because of such things. But what good in nixvim, you are using stylua, so it fixes many things.

Reply:
> `:` is not a magic string. Unfortunately it comes from the vimPlugins in nixpkgs.

In that case I think filtering it is fine, but it'd benefit from an explanatory comment.

Unless either of you disagree?